src/jest.config.integration.js

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line no-undef
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  collectCoverage: true,
+  coverageDirectory: 'coverage/integration',
+  coverageReporters: ['cobertura', 'json', 'html', 'text'],
+  collectCoverageFrom: [
+    './**/*.ts',
+    '!<rootDir>/*/.*\\.types.ts',
+    '!<rootDir>/*/.*\\.mocks.ts',
+    '!<rootDir>/dist/',
+    '!/node_modules/',
+    '!/coverage/',
+  ],
+  coveragePathIgnorePatterns: ['<rootDir>/index-test.ts'],
+  setupFilesAfterEnv: ['./setup-tests.ts'],
+  modulePathIgnorePatterns: ['<rootDir>/dist/'],
+  moduleDirectories: ['node_modules', 'src'],
+  rootDir: './',
+  transform: {
+    '^.+\\.tsx?$': [
+      'ts-jest',
+      {
+        tsconfig: 'tsconfig.json',
+        isolatedModules: true,
+      },
+    ],
+  },
+  transformIgnorePatterns: ['/node_modules/(?!ethereum-cryptography)'],
```

Comment:
is this necessary?

Reply:
I broke down the tests into three separate configurations:

- use different timeouts when running just unit vs (integration or combined)
- use different parallelization when running just unit vs (integration or combined)
- have separate coverage outputs when running individual(unit or integration) or combined

In the end, the CI will consider the combined approach as the total coverage, while you can still run them individually and analyze separately the results